src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/classes/undefined_et.cpp

```diff
@@ -0,0 +1,174 @@
+    auto logical_not = std::make_shared<op::v1::LogicalNot>(cvt_f32);
+
+    function = std::make_shared<ov::Model>(OutputVector{logical_not->output(0)}, ParameterVector{param_0, param_1, param_2}, "UndefinedET");
+}
+
+template<typename TD, typename TS>
+void fill_data(TD* dst, const TS* src, size_t len) {
+    for (size_t i = 0llu; i < len; i++) {
+        dst[i] = static_cast<TD>(src[i]);
+    }
+}
+
+void UndefinedEtSubgraphTest::generate_inputs(const std::vector<ov::Shape>& target_shapes) {
+    inputs.clear();
+    const auto& func_inputs = function->inputs();
+
+#define ET_CASE(P, S, L)                                                           \
+case P :                                                                           \
+fill_data(tensor->data<ov::element_type_traits<P>::value_type>(), S, L); break;
+
+    for (size_t i = 0lu; i < func_inputs.size(); i++) {
+        const auto& param = func_inputs[i];
+        const auto& name = param.get_node()->get_friendly_name();
+        const auto& in_prc = param.get_element_type();
+        std::shared_ptr<ov::Tensor> tensor;
+
+        if (name == "shape") {
+            static const int64_t shape[] = {3, 5, 7};
+            tensor = std::make_shared<ov::Tensor>(in_prc, Shape{3});
+            switch (in_prc) {
+                ET_CASE(element::i32, shape, 3)
+                ET_CASE(element::i64, shape, 3)
+                default:
+                    OPENVINO_THROW("RandomUniform does not support precision ", in_prc, " for the Shape input.");
+            }
+        } else if (name == "minval") {
+            static const float min_val = 0.f;
+            tensor = std::make_shared<ov::Tensor>(in_prc, target_shapes[i]);
+            switch (in_prc) {
+                ET_CASE(ElementType::f32,  &min_val, 1)
+                ET_CASE(ElementType::f16,  &min_val, 1)
+                ET_CASE(ElementType::bf16, &min_val, 1)
+                default:
+                    OPENVINO_THROW("RandomUniform does not support precision ", in_prc, " for the Minval input.");
+            }
+        } else if (name == "maxval") {
+            static const float max_val = 20.f;
+            tensor = std::make_shared<ov::Tensor>(in_prc, target_shapes[i]);
+            switch (in_prc) {
+                ET_CASE(ElementType::f32,  &max_val, 1)
+                ET_CASE(ElementType::f16,  &max_val, 1)
+                ET_CASE(ElementType::bf16, &max_val, 1)
+                default:
+                    OPENVINO_THROW("RandomUniform does not support precision ", in_prc, " for the Maxval input.");
+            }
+        }
+
+#undef ET_CASE
+
+        inputs.insert({param.get_node_shared_ptr(), *tensor});
+    }
+}
+
+TEST_P(UndefinedEtSubgraphTest, CompareWithRefs) {
+    // Disable skipping of the bf16 test cases.
+    const auto origin_sk = ov::test::utils::disable_tests_skipping;
+    ov::test::utils::disable_tests_skipping = true;
```

Comment:
Nit. Do we really need this skip? BF16 is generaly not expected to be run on platforms without avx512 support.

Reply:
Modified